src/test/java/org/broadinstitute/consent/http/resources/TDRResourceTest.java

```diff
@@ -98,4 +115,51 @@ public void testGetDatasetByIdentifier404() {
         assertEquals(404, r.getStatus());
     }
 
+    @Test
+    public void testCreateDraftDataAccessRequest() {
+
+        String identifiers = "DUOS-00001, DUOS-00002";
+
+        Dataset d1 = new Dataset();
+        d1.setDataSetId(1);
+
+        Dataset d2 = new Dataset();
+        d2.setDataSetId(2);
+
+        when(datasetService.findDatasetByIdentifier("DUOS-00001")).thenReturn(d1);
+        when(datasetService.findDatasetByIdentifier("DUOS-00002")).thenReturn(d2);
+
+        initResource();
+
+        Response r = resource.createDraftDataAccessRequest(new AuthUser(), uriInfo, identifiers, "New Project");
+
+        assertEquals(201, r.getStatus());
+    }
+
+
+    @Test
+    public void testCreateDraftDataAccessRequest500() {
+        String identifiers = "DUOS-00001, DUOS-00002";
+
+        when(datasetService.findDatasetByIdentifier("DUOS-00001")).thenReturn(null);
+        when(datasetService.findDatasetByIdentifier("DUOS-00002")).thenReturn(null);
+
+        initResource();
+
+        Response r = resource.createDraftDataAccessRequest(new AuthUser(), uriInfo, identifiers, "New Project");
+
+        assertEquals(HttpStatusCodes.STATUS_CODE_SERVER_ERROR, r.getStatus());
+    }
+
+    @Test
+    public void testFindOrCreateUser() {
+        initResource();
+
+        User tdrUser = resource.findOrCreateUser(new AuthUser());
+        assertNotNull(tdrUser);
+
+        doThrow(new NotFoundException()).when(userService).findUserByEmail(any());
+        User newTdrUser = resource.findOrCreateUser(any());
+        assertNotNull(newTdrUser);
+    }
```

Comment:
As `resource.findOrCreateUser` currently stands, there are two service class methods you would need to mock out, `userService.findUserByEmail` and `userService.createUser`. You would also want to test out the two cases, one where the user is found, and one where the user isn't found. To do that, look at usages of mockito spy and verify in the current codebase.

Reply:
Created a new UserService method to handle this. Added tests for both user found, user not found cases and using spy/verify to check. That was a great exercise for me to learn about those Mockito methods, thanks!